adet/data/augmentation.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+from fvcore.transforms import transform as T
+
+from detectron2.data.transforms import RandomCrop, StandardAugInput
+from detectron2.structures import BoxMode
+
+
+class InstanceAugInput(StandardAugInput):
+    """
+    Keep the old behavior of instance-aware augmentation
+    """
+
+    def __init__(self, *args, **kwargs):
+        instances = kwargs.pop("instances", None)
+        super().__init__(*args, **kwargs)
+        if instances is not None:
+            self.instances = instances
```

Comment:
Note that in theory this implementation will be incorrect if you have >1 augmentations that need to read `instances`. For example, assuming there are two instance-aware cropping, the second cropping will read the same `instances` as the first one, despite that the boxes and everything should've been changed by the first cropping.

This situation is very very rare in practice, though.

Implement `transform()` to property transform `self.instances` would solve that, but transforming `self.instances` is non-trivial: it requires information about the specific dataset (e.g. keypoint flipping) and the strategy can be different depend on custom data in `self.instances`. This is the main reason why we only put primitive types such as `boxes` in `AugInput`.

Reply:
Thank you for letting us know this! We will remove `instances` and use `boxes` for the cropping instead.

Reply:
a fix has just been made: https://github.com/aim-uofa/AdelaiDet/pull/185